Terminal.Gui/Application.cs

```diff
@@ -975,6 +991,8 @@ private static void OnNotifyStopRunState (Toplevel top)
         }
     }
 
+    private static Toplevel _latestClosedRunStateToplevel;
```

Comment:
> 1. Move the declaration above the 1st usage.
>
Done.

> 2. I think it may be more clear as:
> 
> ```cs
> // When `End ()` is called, it is possible `RunState.Toplevel` is a different object than `Top`. This variable is set in `End` in this case so that `Begin` correctly sets `Top`.
> private static Toplevel _cachedRunStateToplevel;
> ```
> 
Done.
> 3. I don't see `_latestClosedRunStateTopLevel` being set to `null` anywhere (e.g. in `ResetState`). Where do you ensure that it's not left set to something?

I already had a commit that fix, but I didn't had pushed yet.